streaming-connect-sink/src/test/java/com/adobe/platform/streaming/integration/AEPSinkConnectorErrorReporterTest.java

```diff
@@ -87,7 +88,39 @@ public void kafkaErrorReporterTest() throws HttpException, IOException, Interrup
 
     // Verify inlet endpoint received 1 XDM record
     getWiremockServer().verify(postRequestedFor(urlEqualTo(getRelativeUrl()))
-      .withRequestBody(equalToJson(payloadReceivedXdmData())));
+      .withRequestBody(equalToJson(payloadReceivedXdmData(XDM_PAYLOAD_FILE))));
+  }
+
+  @Test
+  public void kafkaErrorReporterMultiMessageTest() throws HttpException, IOException, InterruptedException {
+    inletMultiStatusSuccessfulResponse();
+    getConnect().kafka().createTopic(TOPIC_NAME, TOPIC_PARTITION);
+
+    // Create error topic to dump failed data
+    Map<String, String> connectorConfig = connectorConfig(AEP_KAFKA_ERROR_CONNECTOR_CONFIG);
+    getConnect().kafka().createTopic(connectorConfig.get(DEAD_LETTER_TOPIC), TOPIC_PARTITION);
+
+    LOG.info("Starting connector cluster with connector : {}", CONNECTOR_NAME);
+    getConnect().configureConnector(CONNECTOR_NAME, connectorConfig);
+
+    String xdmData = xdmData(XDM_MULTI_MESSAGE_PAYLOAD_FILE);
+    getConnect().kafka().produce(TOPIC_NAME, xdmData);
+    waitForConnectorStart(CONNECTOR_NAME, 1, 8000);
+
+    // Check if error record sent to error topic
+    ConsumerRecords<byte[], byte[]> consumerRecords = getConnect().kafka()
+      .consume(1, 8000, connectorConfig.get(DEAD_LETTER_TOPIC));
+
+    Assertions.assertEquals(1, consumerRecords.count());
+
+    ConsumerRecord<byte[], byte[]> consumerRecord = consumerRecords.iterator().next();
```

Comment:
Can you please add following properties to connector config and test if headers have all error information - 
```
"errors.deadletterqueue.context.headers.enable": "true",
  "errors.log.enable": "true"
  ```